File: camouflage-mgh-hys-lyh/kotlin/src/test/kotlin/camouflage/mgh/hys/lyh/camouflageTest.kt
package camouflage.mgh.hys.lyh

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

fun solution(clothes: Array<Array<String>>) =
        clothes
                .asSequence()
                .groupBy { it.last() }
                .map { it.value.size }
                .fold(1) { acc, i -> acc * (i + 1) }
                .let { it - 1 }



class camouflageTest {

    @Test
    fun `조합할 수 있는 옷의 경우의 수 구하기`() {
        assertThat(solution(arrayOf(
                arrayOf("yellow_hat", "headgear"),
                arrayOf("blue_sunglasses", "eyewear"),
                arrayOf("green_turban", "headgear")
        ))).isEqualTo(5)
    }
}

File: camouflage-mgh-hys-lyh/js/camouflage.test.js
const solution = (arr) =>  Object.values(groupBy(arr)).reduce((acc, cur) => acc * (cur + 1),1) - 1;


const groupBy = (arr) => {
    arr.reduce((acc,v) => ({
        ...acc,
        [v[1]]: acc[v[1]] ? acc[v[1]] + 1 : 1,
    }), {});
}
/*
1. object를 생성.
2. arr 순회하며 의류의 종류를 뽑아 object에 key로 존재하는지 확인
    - 존재하지 않음 : 
    - 존재 : 
*/



test('solution', () => {
    expect(solution([["yellow_hat", "headgear"], ["blue_sunglasses", "eyewear"], ["green_turban", "headgear"]])).toBe(5);
});